### ISSUE TITLE
`no-array-for-each`: False positive with non-arrays

### ISSUE THREAD

**Issue by RebeccaStevens** [sindresorhus/eslint-plugin-unicorn#1788]
Here `CB` is an object with a bunch of functions on it. The `forEach` function has nothing to do arrays but gets flagged.

```js
import * as CB from "strict-callbag-basics";

CB.pipe(
  CB.interval(1000),
  CB.map(x => x + 1),
  CB.filter(x => x % 2),
  CB.take(5),
  CB.forEach(x => console.log(x))
);
```

**Comment by fisker** [comment]
Would you like to send a PR add `CB` to this list?

https://github.com/sindresorhus/eslint-plugin-unicorn/blob/0034e6938a95600288c44a10963506c7bd0cb464/rules/no-array-for-each.js#L373

**Comment by RebeccaStevens** [comment]
Should it also be made so that this list is configurable by the user via an option?

**Comment by fisker** [comment]
https://github.com/sindresorhus/eslint-plugin-unicorn/blob/59218e3cd653fdf9ad7e7f5d2666e8c2fa775601/rules/utils/is-node-matches.js#L46

> Should it also be made so that this list is configurable by the user via an option?

Fine, we can use this https://github.com/sindresorhus/eslint-plugin-unicorn/blob/59218e3cd653fdf9ad7e7f5d2666e8c2fa775601/rules/utils/is-node-matches.js#L46

**Comment by fregante** [comment]
Related, but in that case the issue is just the autofix:

- https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1452
